LICENSE

```diff
@@ -199,3 +199,63 @@
    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
    See the License for the specific language governing permissions and
    limitations under the License.
+
+This repository also contains code from Facebook (from their Fairseq project)
```

Comment:
This is something I highly doubt, because currently we are using:

- From Megatron, since we have transferred the CPP functions to Python, we would not be using anything exactly the same as them. We have simply used them as inspiration, but there are no literal copies.

- What I have not modified to a large extent is `indexed_dataset.py`, originally from Megatron, but they themselves cite Fairseq. Although, as you can see, they also mention that they have rewritten it (The script in Fairseq does indeed resemble it a bit, but it is not a literal copy).

indexed_dataset.py [[NANOTRON](https://github.com/huggingface/nanotron/blob/3dd5f62284d34382789f53f486bd2091d9db2237/src/nanotron/data/indexed_dataset.py)] [[MEGATRON](https://github.com/TJ-Solergibert/Megatron-debug/blob/c5eda947d9728d21b03d77b7db56cb71513d5636/megatron/core/datasets/indexed_dataset.py#L6)] [[FAIRSEQ](https://github.com/facebookresearch/fairseq/blob/bedb259bf34a9fc22073c13a1cee23192fa70ef3/fairseq/data/indexed_dataset.py#L391)]

I moved the LICENSE.